python/lsst/afw/table/catalogMatches.py

```diff
@@ -254,3 +256,112 @@ def copyAliasMapWithPrefix(inSchema, outSchema, prefix=""):
+    catalog : subclass of `lsst.afw.table.BaseCatalog`
+        Catalog to reindex.
+    indices : `numpy.ndarray` of `int`
+        Index array.
+
+    Returns
+    -------
+    new : subclass of `lsst.afw.table.BaseCatalog`
+        Reindexed catalog. Records are shallow copies of those in ``catalog``.
+    """
+    new = type(catalog)(catalog.schema)
+    new.reserve(len(indices))
+    for ii in indices:
+        _rec = catalog[int(ii)]
+        new.extend([_rec], deep=deep)
+    return new
+
+
+def matchCatalogsExact(catalog1, catalog2, patch1=None, patch2=None):
+    """Match two catalogs derived from the same mergeDet catalog
+
+    When testing downstream features, like deblending methods/parameters
+    and measurement algorithms/parameters, it is useful to to compare
+    the same sources in two catalogs. In most cases this must be done
+    by matching on either RA/DEC or XY positions, which occassionally
+    will mismatch one source with another.
+
+    For a more robust solution, as long as the downstream catalog is
+    derived from the same mergeDet catalog, exact source matching
+    can be done via the unique ``(parent, deblend_peakID)``
+    combination. So this function performs this exact matching for
+    all sources both catalogs.
+
+    Parameters
+    ----------
+    catalog1, catalog2: subclass of `lsst.afw.table.BaseCatalog`
+        The two catalogs to merge
+
+    patch1, patch2: array of int
+        Patch for each row, converted into an integer.
+        In the gen3 butler this is done already, in gen2
+        it is recommended to use `patch2Int`, assuming that
+        the patches are the same structure as HSC, that range
+        from '0,0' to '9,9'.
+
+    Returns
+    -------
+    result: tuple of `lsst.afw.table.BaseCatalog`
+        Both catalogs reordered such that all rows are
+        matched to the same source.
+    """
+    # Only match the individual sources, the parents will
+    # already be matched by the mergeDet catalog
+    sidx1 = catalog1["parent"] != 0
+    sidx2 = catalog2["parent"] != 0
+
+    # Create the keys used to merge the catalogs
+    parents1 = np.array(catalog1["parent"][sidx1])
+    peaks1 = np.array(catalog1["deblend_peakId"][sidx1])
+    index1 = np.arange(len(catalog1))[sidx1]
+    parents2 = np.array(catalog2["parent"][sidx2])
+    peaks2 = np.array(catalog2["deblend_peakId"][sidx2])
+    index2 = np.arange(len(catalog2))[sidx2]
+
+    if patch1 is not None:
+        if patch2 is None:
+            msg = "If the catalogs are from different patches then patch1 and patch2 must be specified"
+            msg += ", got {} and {}".format(patch1, patch2)
+            raise ValueError(msg)
+        patch1 = patch1[sidx1]
+        patch2 = patch2[sidx2]
+
+        key1 = np.rec.array((parents1, peaks1, patch1, index1),
+                            dtype=[('parent', np.int64), ('peakId', np.int32),
+                                   ("patch", patch1.dtype), ("index", np.int32)])
```

Comment:
Not at all important, but why do you mix single quotes and double quotes in the same code?

Reply:
Laziness. I must have copied that code from somewhere because I rarely use single quotes, then replaced the first two column names and added more columns myself with double quotes.